File: ui-android-library/src/main/java/com/xyfindables/ui/XYBaseFragment.java
package com.xyfindables.ui;

import android.app.Activity;
import android.app.Fragment;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.res.Resources;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.xyfindables.ui.dialogs.XYThrobberDialog;

import static com.xyfindables.core.XYBase.*;

public class XYBaseFragment extends Fragment {
    final private static String TAG = XYBaseFragment.class.getSimpleName();
    @Nullable
    private ProgressDialog _progressDialog;
    public XYThrobberDialog throbber;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        logInfo(TAG, "onCreate");
        super.onCreate(savedInstanceState);
        throbber = new XYThrobberDialog(getActivity());
    }

    protected void loadViews() {

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        logInfo(TAG, "onCreateView");
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        loadViews();
    }

    public int dpToPx(@NonNull Resources res, int dp) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, res.getDisplayMetrics());
    }

    @Override
    public void onResume() {
        logInfo(TAG, "onResume");
        super.onResume();
    }

    @Override
    public void onPause() {
        logInfo(TAG, "onPause");
        super.onPause();
        throbber.hide();
    }

    protected void showToast(String message) {
        logInfo(TAG, "showToast");
        Toast.makeText(getActivity(), message, Toast.LENGTH_LONG).show();
    }

    private XYThrobberDialog _dialogThrobber = null;

    @Override
    public void onAttach(Context context) {
        logInfo(TAG, "onAttach");
        super.onAttach(context);
    }
}

File: ui-android-library/src/main/java/com/xyfindables/ui/XYBaseFragment.kt
package com.xyfindables.ui

import android.app.Activity
import android.app.ProgressDialog
import android.content.Context
import android.content.res.Resources
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.xyfindables.core.XYBase

import com.xyfindables.ui.dialogs.XYThrobberDialog

open class XYBaseFragment : Fragment() {

    fun logInfo(message: String) {
        XYBase.logInfo(XYBaseFragment.TAG, message)
    }

    fun logExtreme(message: String) {
        XYBase.logExtreme(XYBaseFragment.TAG, message)
    }

    fun logError(message: String, debug: Boolean) {
        XYBase.logError(XYBaseFragment.TAG, message, debug)
    }

    fun logException(exception: Exception, debug: Boolean) {
        XYBase.logException(XYBaseFragment.TAG, exception, debug)
    }

    fun logException(tag: String, exception: Exception, debug: Boolean) {
        XYBase.logException(tag, exception, debug)
    }

    fun logStatus(tag: String, message: String, debug: Boolean) {
        XYBase.logError(tag, message, debug)
    }

    fun logInfo(tag: String, message: String) {
        XYBase.logInfo(tag, message)
    }

    fun logExtreme(tag: String, message: String) {
        XYBase.logExtreme(tag, message)
    }

    fun logError(tag: String, message: String, debug: Boolean) {
        XYBase.logError(tag, message, debug)
    }

    fun logStatus(message: String, debug: Boolean) {
        XYBase.logError(XYBaseFragment.TAG, message, debug)
    }

    private val _progressDialog: ProgressDialog? = null
    var throbber: XYThrobberDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        XYBase.logInfo(TAG, "onCreate")
        super.onCreate(savedInstanceState)
        throbber = XYThrobberDialog(activity as Activity)
    }

    open fun loadViews() {

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        XYBase.logInfo(TAG, "onCreateView")
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        loadViews()
    }

    open fun dpToPx(res: Resources, dp: Int): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp.toFloat(), res.displayMetrics).toInt()
    }

    override fun onResume() {
        XYBase.logInfo(TAG, "onResume")
        super.onResume()
    }

    override fun onPause() {
        XYBase.logInfo(TAG, "onPause")
        super.onPause()
        throbber?.hide()
    }

    protected fun showToast(message: String) {
        XYBase.logInfo(TAG, "showToast")
        Toast.makeText(activity, message, Toast.LENGTH_LONG).show()
    }

    override fun onAttach(context: Context) {
        XYBase.logInfo(TAG, "onAttach")
        super.onAttach(context)
    }

    companion object {
        private val TAG = XYBaseFragment::class.java.simpleName
    }
}

